src/include/btmem.h

```diff
@@ -912,20 +912,13 @@ WT_PACKED_STRUCT_BEGIN(__wt_update)
 
 	WT_UPDATE *next;		/* forward-linked list */
 
-	/*
-	 * Use the maximum size and maximum size-1 as is-deleted and is-reserved
-	 * flags (which means we can't store 4GB objects), instead of increasing
-	 * the size of this structure for a flag bit.
-	 */
-#define	WT_UPDATE_DELETED_VALUE		UINT32_MAX
-#define	WT_UPDATE_DELETED_SET(u)	((u)->size = WT_UPDATE_DELETED_VALUE)
-#define	WT_UPDATE_DELETED_ISSET(u)	((u)->size == WT_UPDATE_DELETED_VALUE)
-
-#define	WT_UPDATE_RESERVED_VALUE	(UINT32_MAX - 1)
-#define	WT_UPDATE_RESERVED_SET(u)	((u)->size = WT_UPDATE_RESERVED_VALUE)
-#define	WT_UPDATE_RESERVED_ISSET(u)	((u)->size == WT_UPDATE_RESERVED_VALUE)
 	uint32_t size;			/* update length */
 
+#define	WT_UPDATE_STANDARD	0
+#define	WT_UPDATE_DELETED	1
+#define	WT_UPDATE_RESERVED	2
+	uint8_t type;			/* update type */
```

Comment:
I think this deserves a comment. I wondered why this didn't use either an enum or a regular flags field. I presume the reason is that the structure is space sensitive so warrants a different approach?